pushnotifications/src/main/java/com/pusher/pushnotifications/reporting/ReportingJobService.kt

```diff
@@ -79,8 +85,14 @@ class ReportingJobService: JobService() {
           )
         }
         ReportEventType.Open -> {
+          // returning `null` if the instance id is missing because it's possible that
+          // we are processing a bundle that was created with an old SDK version that
+          // didn't had this key. Our migration strategy is to drop the reporting
+          // as it's (a) a rare one-time transition and (b) it's a best effort feature.
+          val instanceId = bundle.getString(BUNDLE_INSTANCE_ID_KEY) ?: return null
```

Comment:
Why not to put this line and the comment only once as the first line in this method? It is a common convention to check prerequisites at the beginning of a function. It would be also more legible to have clear if-return statement as opposed to jumping out of the fun from within the elvis operator.

Reply:
> Why not to put this line and the comment only once as the first line in this method?
Because we don't know the type before we parse it. And we could have a type that has `instanceId`. Well, probably all of them will have `instanceId`, so I've changed this in d7b6cda.